contracts/btc-finality/src/contract.rs

```diff
@@ -290,26 +321,34 @@ fn send_rewards_msg(
         })
         .map(|item| {
             let (fp_pubkey_hex, reward) = item?;
-            Ok(babylon_contract::msg::contract::RewardsDistribution {
+            Ok(babylon_apis::btc_staking_api::RewardInfo {
                 fp_pubkey_hex,
                 reward,
             })
         })
         .collect::<StdResult<Vec<_>>>()?;
-    let msg = babylon_contract::ExecuteMsg::SendRewards {
-        fp_distribution: fp_rewards.clone(),
-    };
-    let wasm_msg = WasmMsg::Execute {
-        contract_addr: cfg.babylon.to_string(),
-        msg: to_json_binary(&msg)?,
-        funds: coins(rewards, cfg.denom.as_str()),
+    let wasm_msg = if recipient == cfg.babylon {
+        let msg = babylon_contract::ExecuteMsg::SendRewards {
```

Comment:
```suggestion
        // the reward is sent to the Babylon contract and the Babylon
        // contract will send the reward to Babylon chain for distirbution
        let msg = babylon_contract::ExecuteMsg::SendRewards {
```

Reply:
:+1 This will be changed / deprecated in a follow-up anyway.